django_sharding_library/utils.py

```diff
@@ -0,0 +1,32 @@
+from django.db import connections, DatabaseError, transaction
+from django.conf import settings
+from django_sharding_library.sql import postgres_shard_id_function_sql
+from django.db.models import signals
+
+
+def create_postgres_global_sequence(sequence_name, db_alias, reset_sequence=False):
+    cursor = connections[db_alias].cursor()
+    sid = transaction.savepoint(db_alias)
+    try:
+        cursor.execute("CREATE SEQUENCE %s;" % sequence_name)
+    except DatabaseError:
+        transaction.savepoint_rollback(sid, using=db_alias)
+        if reset_sequence:
+            cursor.execute("SELECT setval('%s', 1, false)" % (sequence_name,))
+    else:
+        transaction.savepoint_commit(sid, using=db_alias)
+    cursor.close()
+
+
+def create_postgres_shard_id_function(sequence_name, db_alias, shard_id):
+    cursor = connections[db_alias].cursor()
+    cursor.execute(postgres_shard_id_function_sql % {'shard_epoch': settings.SHARD_EPOCH,
+                                                     'shard_id': shard_id,
+                                                     'sequence_name': sequence_name})
+    cursor.close()
+
+
+def register_migration_signal_for_model_receiver(model, function, dispatch_uid=None):
+    signals.pre_migrate.connect(function, sender=model, dispatch_uid=dispatch_uid)
```

Comment:
As far as I can find in the documentation, there is no way to force a dependency because of a field when a user run makemigrations. I also asked in the django irc and the response from the admin there at the time was that he didn't think it was possible.

This is exactly why the pre-migration signal was added in though, its for pre-migration processing. I am pretty sure this is the "django" way of doing it short of writing a migration, putting it in the app, and figuring out how to force it as a dependency on another app based on arbitrary, non-built-in logic (for example, foreign key fields have special logic already built into the migration builder). I am not sure I can think of a better use case of the pre migration signal than this